source/beamline/IDEAS/IDEASBeamline.cpp

```diff
@@ -248,3 +248,59 @@ void IDEASBeamline::onShutterStatusChanged()
 {
 	emit overallShutterStatus(safetyShutter_->isOpen() && photonShutter2_->isOpen() && safetyShutter2_->isOpen());
 }
+
+AMAction3 *IDEASBeamline::createSamplePlatformMoveVertical(double verticalPosition)
+{
+	if(!IDEASBeamline::samplePlatformVertical_->isConnected())
+		return 0;
+
+	return AMActionSupport::buildControlMoveAction(samplePlatformVertical_, verticalPosition);
+}
+
+AMAction3 *IDEASBeamline::createSamplePlatformMoveHorizontal(double horiztonalPosition)
+{
+	if(!IDEASBeamline::samplePlatformHorizontal_->isConnected())
+		return 0;
+
+	return AMActionSupport::buildControlMoveAction(samplePlatformHorizontal_, horiztonalPosition);
+}
+
+AMAction3 *IDEASBeamline::createVaccumSampleStageMove(double vaccumStagePosition)
+{
+	if(!IDEASBeamline::vacuumSampleStage_->isConnected())
+		return 0;
+
+	return AMActionSupport::buildControlMoveAction(vacuumSampleStage_, vaccumStagePosition);
+}
+
+AMAction3 *IDEASBeamline::createJJSlitHGapMove(double jjSlitHGapPosition)
+{
+	if(!IDEASBeamline::jjSlitHGap_->isConnected())
+		return 0;
+
+	return AMActionSupport::buildControlMoveAction(jjSlitHGap_, jjSlitHGapPosition);
+}
+
+AMAction3 *IDEASBeamline::createJJSlitHCenterMove(double jjSlitHCenterPosition)
+{
+	if(!IDEASBeamline::jjSlitHCenter_->isConnected())
+		return 0;
+
+	return AMActionSupport::buildControlMoveAction(jjSlitHCenter_, jjSlitHCenterPosition);
+}
+
+AMAction3 *IDEASBeamline::createJJSlitVGapMove(double jjSlitVGapPosition)
+{
+	if(!IDEASBeamline::jjSlitVGap_->isConnected())
+		return 0;
+
+	return AMActionSupport::buildControlMoveAction(jjSlitVGap_, jjSlitVGapPosition);
+}
+
+AMAction3 *IDEASBeamline::createJJSlitVCenterMove(double jjSlitVCenterPosition)
+{
+	if(!IDEASBeamline::jjSlitVCenter_->isConnected())
```

Comment:
Because you are inside of IDEASBeamline already, the ```IDEASBeamline::``` can be removed from all the member names.